scalyr_agent/builtin_monitors/syslog_monitor.py

```diff
@@ -570,77 +576,113 @@ class SyslogUDPHandler(six.moves.socketserver.BaseRequestHandler):
     a protocol neutral handler
     """
 
+    def __init__(self, request_processing_executor, request, client_address, server):
+        self.__request_processing_executor = request_processing_executor
+
+        if six.PY3:
+            super(SyslogUDPHandler, self).__init__(request, client_address, server)
+        else:
+            six.moves.socketserver.BaseRequestHandler.__init__(self, request, client_address, server)
+
+    def factory_method(request_processing_executor):
+        def create_handler(request, client_address, server):
+            return SyslogUDPHandler(request_processing_executor, request, client_address, server)
+
+        return create_handler
+
+
     def handle(self):
         data = six.ensure_text(self.request[0].strip(), "utf-8", errors="ignore")
         extra = {
             "proto": "udp",
             "srcip": self.client_address[0],
             "destport": self.server.server_address[1],
         }
-        self.server.syslog_handler.handle(data, extra)
 
+        self.__request_processing_executor.submit(
```

Comment:
Since `handle` is now called inside a thread, do we need any kind of locking around this method?

It appears we already have a lock in https://github.com/scalyr/scalyr-agent-2/blob/888447321333b7f817739cb06806a461188b3ba7/scalyr_agent/builtin_monitors/syslog_monitor.py#L1583 for file creation related reasons and all `handle()` method does is pass data to the logger (https://github.com/scalyr/scalyr-agent-2/blob/888447321333b7f817739cb06806a461188b3ba7/scalyr_agent/builtin_monitors/syslog_monitor.py#L1738).

I believe we already have a lock in our custom logger methods, but it would be good to confirm there are no race conditions - starting a monitor and sending a lot of data to in in parallel manner and verifying that syslog monitor log file is correct aka that data is not interleaved.

Reply:
Thanks for the clarification :)